src/main/java/sg/ncl/MainController.java

```diff
@@ -121,6 +121,21 @@ public String resources() {
         return "resources";
     }
 
+    @RequestMapping("/research")
+    public String research() {
+        return "research";
+    }
+
+    @RequestMapping("/calendar")
+    public String calendar() {
+        return "calendar";
+    }
+
+    @RequestMapping("/calendar1")
+    public String calendar1() {
+        return "calendar1";
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this method and declare a constant for this value. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3400)